### PR TITLE
Added noise and selecting the best model based on accuracy

### PR DESCRIPTION
**Pull request by arsalananwar11**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -75,15 +91,15 @@ def train(model, iterator, optimizer, criterion, device, scheduler=None):
             
     return epoch_loss / len(iterator), correct / total
 
-def test(model, iterator, criterion, device):
-    
+def test(model, iterator, criterion, device, epoch, args):
+    global best_acc
```
Comment (tkobil) [inline main.py:95]: let's avoid global variables here. we can probably pass in to the test method if needed

File: main.py [hunk 2]
```diff
@@ -27,20 +40,23 @@ def get_optimizer(model, optimizer_type, lr, momentum, weight_decay):
...
+        return optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=200)
     
 def get_transform(transform):
     if transform:
         return torchvision.transforms.Compose([
             torchvision.transforms.RandomCrop(32, padding=4),
             torchvision.transforms.RandomHorizontalFlip(),
             torchvision.transforms.ToTensor(),
-            torchvision.transforms.Normalize((0.4915, 0.4823, .4468), (0.2470, 0.2435, 0.2616)), # mean/std
+            #torchvision.transforms.Normalize((0.4915, 0.4823, .4468), (0.2470, 0.2435, 0.2616)), # mean/std
```
Comment (tkobil) [inline main.py:57]: feel free to delete commented out code
